sigle/src/modules/layout/components/HeaderDropdown.tsx

```diff
@@ -0,0 +1,246 @@
+import Link from 'next/link';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  Button,
+  Box,
+  Typography,
+  StyledChevron,
+  Flex,
+} from '../../../ui';
+import { generateAvatar } from '../../../utils/boringAvatar';
+import {
+  createNewEmptyStory,
+  getStoriesFile,
+  saveStoriesFile,
+  saveStoryFile,
+} from '../../../utils';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+import { useQueryClient } from 'react-query';
+import { useGetUserSettings } from '../../../hooks/appData';
+import * as Fathom from 'fathom-client';
+import { Goals } from '../../../utils/fathom';
+import { userSession } from '../../../utils/blockstack';
+import { createSubsetStory } from '../../editor/utils';
+import { styled } from '../../../stitches.config';
+import { useAuth } from '../../auth/AuthContext';
+import { signOut } from 'next-auth/react';
+import {
+  ArchiveIcon,
+  CrumpledPaperIcon,
+  FileTextIcon,
+  MixIcon,
+} from '@radix-ui/react-icons';
+import { useTheme } from 'next-themes';
+import { Switch, SwitchThumb } from '../../../ui/Switch';
+import { useGetUserSubscription } from '../../../hooks/subscriptions';
+
+const ImageContainer = styled('div', {
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  overflow: 'hidden',
+  width: 24,
+  height: 24,
+  br: '$1',
+});
+
+export const HeaderDropdown = () => {
+  const { data: settings } = useGetUserSettings();
+  const router = useRouter();
+  const { user } = useAuth();
+  const queryClient = useQueryClient();
+  const [loadingCreate, setLoadingCreate] = useState(false);
+  const { resolvedTheme, setTheme } = useTheme();
+  const { data: userSubscription } = useGetUserSubscription();
```

Comment:
This will fail for legacy users, can we trigger the query only for non legacy users?